.github/workflows/main.yaml

```diff
@@ -2,87 +2,121 @@ name: Build, push and deploy
 on: push
 env:
   IMAGE_TAG: ${{ github.sha }}
-  IMAGE: docker.pkg.github.com/${{ github.repository }}/veilarbperson
+  IMAGE: ghcr.io/${{ github.repository }}/veilarbperson
   PRINT_PAYLOAD: true
 jobs:
+  link-to-trello:
+    runs-on: ubuntu-latest
+    name: Trello update
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Get trello card id
+        id: card
+        uses: navikt/veilarbtrelloactions/get-card@v2.0
+        with:
+          key: ${{ secrets.TRELLO_KEY }}
+          token: ${{ secrets.TRELLO_TOKEN }}
+          board: BcjrjQZk
+      - name: Attach branch to card
+        id: Attach
+        uses: navikt/veilarbtrelloactions/create-attachment@v2.0
+        if: ${{ steps.card.outputs.success }}
+        with:
+          key: ${{ secrets.TRELLO_KEY }}
+          token: ${{ secrets.TRELLO_TOKEN }}
+          card-id: ${{ steps.card.outputs.card-id }}
+          attachment-type: branch
   test:
     name: Run tests
     runs-on: ubuntu-latest
-    if: github.ref != 'refs/heads/dev' && github.ref != 'refs/heads/master'
+    if: github.ref != 'refs/heads/master'
     steps:
       - name: Checkout
-        uses: actions/checkout@v1
-      - name: Set up JDK 11
-        uses: actions/setup-java@v1
+        uses: actions/checkout@v2
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
         with:
-          java-version: 11
+          java-version: 17
+          distribution: 'temurin'
       - name: Set up cache
         uses: actions/cache@v1
         with:
           path: ~/.m2/repository
           key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-maven-
       - name: Run maven tests
+        env:
+          MAVEN_OPTS: -Xss1024M -Xmx2048M
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: mvn -B verify
-
-  test-build-and-push:
-    name: Test, build and push
+  build-and-push:
+    name: Build and push
     runs-on: ubuntu-latest
     if: github.ref == 'refs/heads/dev' || github.ref == 'refs/heads/master'
     steps:
       - name: Checkout
-        uses: actions/checkout@v1
-      - name: Set up JDK 11
-        uses: actions/setup-java@v1
+        uses: actions/checkout@v3
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
         with:
-          java-version: 11
+          java-version: 17
+          distribution: 'temurin'
       - name: Set up cache
-        uses: actions/cache@v1
+        uses: actions/cache@v3
         with:
           path: ~/.m2/repository
           key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-maven-
       - name: Build maven artifacts
-        run: mvn -B package
+        run: mvn -Dgithub.token=${{ secrets.GITHUB_TOKEN }} -B package -D skipTests
```

Comment:
Is -Dgithub.token=${{ secrets.GITHUB_TOKEN }} necessary?

Reply:
I will try without it